zerver/lib/markdown/__init__.py

```diff
@@ -1826,17 +1845,27 @@ def handleMatch(self, m: Match[str]) -> Optional[Element]:
                 el.set("class", "user-mention")
                 text = f"@{text}"
             el.text = markdown.util.AtomicString(text)
-            return el
-        return None
-
+            return el, match_span[0], match_span[1]
+        return None, None, None
 
-class UserGroupMentionPattern(markdown.inlinepatterns.Pattern):
-    def handleMatch(self, m: Match[str]) -> Optional[Element]:
-        match = m.group(2)
 
+class UserGroupMentionPattern(markdown.inlinepatterns.InlineProcessor):
+    def handleMatch(  # type: ignore[override] # supertype incompatible with supersupertype
+        self, m: Match[str], data: str
+    ) -> Union[Tuple[None, None, None], Tuple[Element, int, int]]:
+        name = m.group(0)
+        match_span = m.span()
         db_data = self.md.zulip_db_data
+
+        # As patterns like @*group name* are captured.
+        start_syntax = "@*"
+        end_syntax = "*"
         if self.md.zulip_message and db_data is not None:
-            name = extract_user_group(match)
+            # As InlineProcessor API sends all the matching part
+            # coming after (?<![^\s\'\"\(,:<]), so @ is also
+            # included in name so it needs to be strip.
+            if name.startswith(start_syntax) and name.endswith(end_syntax):
+                name = name[len(start_syntax) : -len(end_syntax)]
```

Comment:
Why not `name = extract_user_group(m[1])`?

Reply:
extract_user_group(m[1]) will perform name = name[1:-1], but we have name="@*caught_pattern*" initially, so we need name[2:-1]. So I extracted it here instead of making a new function, or putting a flag in the current function.

Reply:
Note `m[1]`, not `m[0]`. Group 1 is the part *after* the `@`. (That’s why `extract_user_group` works at all.)

Reply:
Made these changes.